api/include/opentelemetry/trace/propagation/http_trace_context.h

```diff
@@ -0,0 +1,265 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <array>
+#include <iostream>
+#include <map>
+#include <string>
+#include "opentelemetry/context/context.h"
+#include "opentelemetry/nostd/shared_ptr.h"
+#include "opentelemetry/nostd/span.h"
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/nostd/variant.h"
+#include "opentelemetry/trace/default_span.h"
+#include "opentelemetry/trace/key_value_iterable.h"
+#include "opentelemetry/trace/propagation/http_text_format.h"
+#include "opentelemetry/trace/span.h"
+#include "opentelemetry/trace/span_context.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace trace
+{
+namespace propagation
+{
+static const nostd::string_view kTraceParent = "traceparent";
+static const nostd::string_view kTraceState  = "tracestate";
+static const int kTraceDelimiterBytes        = 3;
+static const int kHeaderElementLengths[4]    = {2, 32, 16, 2};
```

Comment:
This is confusing now. It's important to know what these numbers indicate - which is getting missed now. Better to have constants as defined earlier, and use them within `kHeaderElementLengths` array. Also, size of trace_id is 16 bytes, and span_id is 8 bytes. Why are we using 32 bytes and 16 bytes here ?

Reply:
I just followed a recommendation to remove the definitions and convert them into this form though. I could provide comments on the size to bolster understanding.

Reply:
Assuming you're referring to https://github.com/open-telemetry/opentelemetry-cpp/pull/143#discussion_r476098485, there wasn't a recommendation to remove the definitions. reyang pointed out that the numbers are repeated, but didn't say "get rid of the constants".

I agree with lalitb: the constants give useful mnemonics to the values.


Regarding why the values are {2,32,16,2} - these are characters in a hex string, which are nibbles not bytes. They're 2:1 with the byte size.

Reply:
So I just added those constants back but this time with those constants using the values stored in kHeadeElementLengths as definition.